### PR TITLE
add all missing types from wiki.vg

### PR DESCRIPTION
**Pull request by u9g**
<details>
<summary>code used to make this:</summary>
<br>

```js
const fs = require('fs')
const path = require('path')
const protocol = require('./protocol.json')
const t = {
  byte: 'i8',
  short: 'i16',
  int: 'i32',
  long: 'i64',
  float: 'f32',
  double: 'f64',
  unsigned_byte: 'u8',
  unsigned_short: 'u16',
  unsigned_int: 'u32',
  unsigned_long: 'u64',
  string: 'string',
  varint: 'varint',
  byte_array: 'byte_array',
  vec3: 'vec3',
  uvcoords: 'uvcoords',
  animData: 'animData'
}
function addPacket (protocol, _newPacket) {
  if (_newPacket.way === 'toClient' || _newPacket.way === 'bidirectional') {
    protocol.toClient.types.packet[1][0].type[1].mappings[_newPacket.id] = _newPacket.name
    protocol.toClient.types.packet[1][1].type[1].fields[_newPacket.name] = `packet_${_newPacket.name}`
    protocol.toClient.types[`packet_${_newPacket.name}`] = [
      'container',
      Object.entries(_newPacket.params).map((elem) => ({ name: elem[0], type: elem[1] }))
    ]
  }
  if (_newPacket.way === 'toServer' || _newPacket.way === 'bidirectional') {
    protocol.toServer.types.packet[1][0].type[1].mappings[_newPacket.id] = _newPacket.name
    protocol.toServer.types.packet[1][1].type[1].fields[_newPacket.name] = `packet_${_newPacket.name}`
    protocol.toServer.types[`packet_${_newPacket.name}`] = [
      'container',
      Object.entries(_newPacket.params).map((elem) => ({ name: elem[0], type: elem[1] }))
    ]
  }
  Object.values(_newPacket.params).forEach(el => { protocol.types[el] = 'native' })
}
function addTypes (protocol, _types) {
  Object.entries(_types).forEach(elem => {
    protocol.types[elem[0]] = ['container', Object.entries(elem[1]).map(data => ({ name: data[0], type: data[1] }))]
  })
}
const packets = [
  {
    name: 'two_way_ping',
    way: 'bidirectional',
    id: '0x2B',
    params:
        {
          direction: t.byte,
          data: t.short
        }
  },
  {
    name: 'env_set_weather_type',
    way: 'toClient',
    id: '0x1F',
    params:
      {
        weatherType: 'byte'
      }
  },
  {
    name: 'set_inventory_order',
    way: 'toClient',
    id: '0x2C',
    params: {
      order: t.unsigned_byte,
      blockId: t.unsigned_byte
    }
  },
  {
    name: 'set_hotbar',
    way: 'toClient',
    id: '0x2D',
    params: {
      blockId: t.unsigned_byte,
      hotbarIndex: t.unsigned_byte
    }
  },
  {
    name: 'set_spawn_point',
    way: 'toClient',
    id: '0x2E',
    params: {
      spawnX: t.short,
      spawnY: t.short,
      spawnZ: t.short,
      spawnYaw: t.byte,
      spawnPitch: t.byte
    }
  },
  {
    name: 'velocity_control',
    way: 'toClient',
    id: '0x2F',
    params: {
      velocityX: t.int,
      velocityY: t.int,
      velocityZ: t.int,
      modeX: t.byte,
      modeY: t.byte,
      modeZ: t.byte
    }
  },
  {
    name: 'define_effect',
    way: 'toClient',
    id: '0x30',
    params: {
      effectId: t.unsigned_byte,
      u1: t.unsigned_byte,
      v1: t.unsigned_byte,
      u2: t.unsigned_byte,
      v2: t.unsigned_byte,
      redTint: t.unsigned_byte,
      greenTint: t.unsigned_byte,
      blueTint: t.unsigned_byte,
      frameCount: t.unsigned_byte,
      particleCount: t.unsigned_byte,
      size: t.unsigned_byte,
      sizeVariation: t.int,
      spread: t.unsigned_short,
      speed: t.int,
      gravity: t.int,
      baseLifetime: t.int,
      lifetimeVariation: t.int,
      collideFlags: t.unsigned_byte,
      fullBright: t.byte
    }
  },
  {
    name: 'spawn_effect',
    way: 'toClient',
    id: '0x31',
    params: {
      effectId: t.byte,
      positionX: t.int,
      positionY: t.int,
      positionZ: t.int,
      originX: t.int,
      originY: t.int,
      originZ: t.int
    }
  },
  {
    name: 'define_model',
    way: 'toClient',
    id: '0x32',
    params: {
      modelId: t.unsigned_byte,
      name: t.string,
      flags: t.unsigned_byte,
      nameY: t.float,
      eyeY: t.float,
      collisionSize: t.vec3,
      pickingBoundsMin: t.vec3,
      pickingBoundsMax: t.vec3,
      uScale: t.unsigned_short,
      vScale: t.unsigned_short,
      partsCount: t.unsigned_byte
    }
  },
  {
    name: 'define_model_part',
    way: 'toClient',
    id: '0x32',
    params: {
      modelId: t.unsigned_byte,
      minCoords: t.vec3,
      maxCoords: t.vec3,
      topFaceUV: t.uvcoords,
      bottomFaceUV: t.uvcoords,
      frontFaceUV: t.uvcoords,
      backFaceUV: t.uvcoords,
      leftFaceUV: t.uvcoords,
      rightFaceUV: t.uvcoords,
      rotationOrigin: t.vec3,
      rotationAngles: t.vec3,
      animationOne: t.animData,
      animationTwo: t.animData,
      animationThree: t.animData,
      animationFour: t.animData,
      flags: t.unsigned_byte
    }
  }
]
const types = {
  vec3: {
    x: t.float,
    y: t.float,
    z: t.float
  },
  uvcoords: {
    u1: t.unsigned_short,
    v1: t.unsigned_short,
    u2: t.unsigned_short,
    v2: t.unsigned_short
  },
  animData: {
    flags: t.byte,
    a: t.float,
    b: t.float,
    c: t.float,
    d: t.float
  }
}

const newProtocol = JSON.parse(JSON.stringify(protocol))
packets.forEach(packet => addPacket(newProtocol, packet))
addTypes(newProtocol, types) // declare types after so the ones thats had native put are overwritten
fs.writeFileSync(path.join(__dirname, 'protocol.json'), JSON.stringify(newProtocol, null, 2))

```
</details>

### CODE REVIEW

File: data/protocol.json [hunk 1]
```diff
@@ -97,7 +164,8 @@
                   "0x22": "player_clicked",
                   "0x10": "ext_info",
                   "0x11": "ext_entry",
-                  "0x13": "custom_block_support_level"
+                  "0x13": "two_way_ping",
```
Comment (rom1504) [inline data/protocol.json]: looks wrong, duplicate packet

**Comment by rom1504** [comment]
does it run ?

File: data/protocol.json (hunk 1 above)
Comment (u9g) [reply data/protocol.json]: oh oops, 1 sec, thought I fixed this

Comment (u9g) [reply data/protocol.json]: fixed

**Comment by u9g** [comment]
> does it run ?

yup

**Review by rom1504 — APPROVED** [review]

**Pull request merged**